src/main/java/slave/elements/Parser.java

```diff
@@ -0,0 +1,139 @@
+package slave.elements;
+
+import slave.command.AddToDoCommand;
+import slave.command.NullCommand;
+import slave.command.ListCommand;
+import slave.command.HelpCommand;
+import slave.command.ExitCommand;
+import slave.command.DoneCommand;
+import slave.command.DeleteCommand;
+import slave.command.ClearCommand;
+import slave.command.AddEventCommand;
+import slave.command.Command;
+import slave.command.AddDeadlineCommand;
+import slave.command.FindCommand;
+
+import slave.exception.DukeException;
+import slave.exception.MissingDateException;
+import slave.exception.MissingDescriptionException;
+import slave.exception.MissingTaskException;
+
+/**
+ * A parser to take in input by user and returns the appropriate command.
+ */
+public class Parser {
+
+    /**
+     * Static method which parses input by user and returns the corresponding command object.
+     *
+     * @param fullCommand Input by user.
+     * @return Corresponding command based on user input.
+     * @throws DukeException For invalid input.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Following the logic of your code here, would it be better to remove your enum class entirely?